src/main/scala/uk/gov/nationalarchives/tdr/api/db/DbConnection.scala

```diff
@@ -1,7 +1,57 @@
 package uk.gov.nationalarchives.tdr.api.db
 
-import akka.stream.alpakka.slick.javadsl.SlickSession
+import akka.stream.alpakka.slick.scaladsl.SlickSession
+import com.typesafe.config.ConfigFactory
+import scalacache.CacheConfig
+import scalacache.caffeine.CaffeineCache
+import scalacache.memoization._
+import scalacache.modes.try_._
+import slick.jdbc.JdbcBackend
+import slick.jdbc.hikaricp.HikariCPJdbcDataSource
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.rds.RdsUtilities
+import software.amazon.awssdk.services.rds.model.GenerateAuthenticationTokenRequest
+
+import scala.concurrent.duration._
+import scala.util.{Failure, Success, Try}
 
 object DbConnection {
-  val db = SlickSession.forConfig("consignmentapi").db
+  implicit val passwordCache: CaffeineCache[String] = CaffeineCache[String](CacheConfig())
+  val slickSession: SlickSession = SlickSession.forConfig("consignmentapi")
+
+  def db: JdbcBackend#DatabaseDef = {
+    val db = slickSession.db
+    db.source match {
+      case hikariDataSource: HikariCPJdbcDataSource =>
+        val configBean = hikariDataSource.ds.getHikariConfigMXBean
+        getPassword match {
+          case Failure(exception) => throw exception
+          case Success(password) =>
+            configBean.setPassword(password)
+            db
+        }
+      case _ =>
+        db
+    }
+  }
+
+  def getPassword: Try[String] = memoize[Try, String](Some(5.minutes)) {
+    val configFactory = ConfigFactory.load
+    val useIamAuth = configFactory.getBoolean("consignmentapi.useIamAuth")
+    if (useIamAuth) {
+      val rdsClient = RdsUtilities.builder().region(Region.EU_WEST_2).build()
+      val port = 5432
+      val request = GenerateAuthenticationTokenRequest.builder()
+        .credentialsProvider(DefaultCredentialsProvider.builder().build())
+        .hostname(sys.env("DB_ADDR"))
```

Comment:
Could this be read from an environment variable in the config, like the rest of the env vars?

Reply:
Yeah makes sense. The only reason I didn't originally was because there was no parameter in application.conf with just this value, it was used as part of the url. I've changed application.conf to have a host property which I'm reading in here.